sk0-front/src/app/auth/auth.service.ts

```diff
@@ -25,72 +26,96 @@ export class AuthService {
+      next: async (user: LoginResponse) => {
+        this._userName = user.name;
+        this._isUserLoggedIn.next(true);
+        await this.showSuccessMessage('Sikeres bejelentkezés');
+        this.navCtrl.navigateForward('/home');
+      },
+      error: (response: HttpErrorResponse) => {
+        let errorMessage: string;
+  
+        switch (response.status) {
+          case 400:
+            errorMessage = 'Érvénytelen kérés';
+            break;
+          case 401:
+            errorMessage = 'Nem megfelelő email cím vagy jelszó';
+            break;
+          case 403:
+            errorMessage = 'Hozzáférés megtagadva';
+            break;
+          case 404:
+            errorMessage = 'Az oldal nem található';
+            break;
+          case 500:
+            errorMessage = 'Belső szerverhiba';
+            break;
+          default:
+            errorMessage = 'Ismeretlen hiba történt';
+            break;
+        }
+  
+        this.showError(errorMessage);
       }
-    } catch (error) {
-      const errorMessage = (error as Error).message;
-      return this.showError(errorMessage);
-    }
-    this._userName = this.mockUsers.find((user) => user.email === email)!.name;
-    this._isUserLoggedIn = true;
+    });
   }
 
   async register(registerFormData: {
     name: string;
     email: string;
     password: string;
   }) {
-    const { email, password } = registerFormData;
-    // TODO: Connect FE with BE and remove mockUsers and refactor code below
-    const foundUser = this.mockUsers.find((user) => {
-      return user.email === email;
-    });
-    try {
-      if (foundUser !== undefined) {
-        throw new Error('A felhasználó már létezik.');
-      }
-      if (password.length < 6) {
-        throw new Error('A jelszó túl gyenge.');
+    const { name, email, password } = registerFormData;
+  
+    this.http.post(`${environment.baseUrl}/users/register`, { name, email, password }).subscribe({
+      next: async () => {
+        this._userName = name;
+        await this.showSuccessMessage('Sikeres regisztráció');
+        this.navCtrl.navigateForward('/login');
+      },
+      error: (response: HttpErrorResponse) => {
+        let errorMessage: string;
+  
+        switch (response.status) {
+          case 400:
+            errorMessage = 'Érvénytelen kérés';
+            break;
+          case 401:
+            errorMessage = 'Nem megfelelő email cím vagy jelszó';
+            break;
+          case 403:
+            errorMessage = 'Hozzáférés megtagadva';
+            break;
+          case 404:
+            errorMessage = 'Az oldal nem található';
+            break;
+          case 409:
+            errorMessage = 'Már van felhasználó ezzel az e-mail címmel.';
+            break;
+          case 500:
+            errorMessage = 'Belső szerverhiba';
+            break;
+          default:
+            errorMessage = 'Ismeretlen hiba történt';
+            break;
+        }
```

Comment:
What do you think should we rather use a map instead of switch-case? It will help us later with the internalization too.

Reply:
I like this solution, I change it to this